pkg/convert/xliff2.go

```diff
@@ -145,6 +140,116 @@ func ToXliff2(translation model.Translation) ([]byte, error) {
 
 // helpers
 
+// messageFromUnit converts Xliff2.0 unit element to model.Message.
+func messageFromUnit(u unit, original bool) (*model.Message, error) {
+	var (
+		err     error
+		decoder *xml.Decoder
+	)
+
+	m := &model.Message{
+		ID:          u.ID,
+		Description: descriptionsFromXliff2(u),
+		Positions:   positionsFromXliff2(u.Notes),
+	}
+
+	if original {
+		decoder = xml.NewDecoder(bytes.NewBufferString(u.Source.Content))
+		m.Status = model.MessageStatusTranslated
+	} else {
+		decoder = xml.NewDecoder(bytes.NewBufferString(u.Target.Content))
+		m.Status = model.MessageStatusUntranslated
+	}
+
+	// content to MF2 message
+
+	elementCount := make(map[string]int)
+	message := mf2.NewBuilder()
+
+	if !original {
+		// NOTE: source text is stored for use in MF2 to XLIFF2.0 conversion.
+		message.Local("$source", mf2.Literal("<source>"+u.Source.Content+"</source>"))
+	}
+
+	for {
+		var token xml.Token
+
+		token, err = decoder.Token()
+		if errors.Is(err, io.EOF) {
+			break
+		} else if err != nil {
+			return nil, fmt.Errorf("get token: %w", err)
+		}
+
+		if token == nil {
+			break
+		}
+
+		switch t := token.(type) {
+		case xml.CharData: // Text
+			message.Text(unescapeXML(string(t)))
+		case xml.StartElement:
+			switch elName := t.Name.Local; elName {
+			default:
+				// noop
+			case cp, ph, sc, ec, sm, em, pc, mrk:
+				elementCount[elName]++
+
+				v := "$" + elName + strconv.Itoa(elementCount[elName])
+				message.Expr(mf2.Var(v))
+
+				var attributes string
+
+				for i := range t.Attr {
+					attributes += fmt.Sprintf(` %s="%s"`, t.Attr[i].Name.Local, t.Attr[i].Value)
```

Comment:
Why not marshalling using XML? Are you sure it always serializes correctly to XML?

Reply:
Why do we store XML syntax in MF2?

Reply:
Why do we store XML syntax in MF2? 
Followed design document that @VladislavsPerkanuks  drew up. Maybe he can elaborate.

Reply:
in MF2 we store everything, that is related to message and is important, and cannot fit into model.Message struct.
So we store them as locals. When converting back to xliff, we get that info from locals, and store them back into xliff struct.

That was the way, that @janishorsts and me agreed on.